### DISCUSSION TITLE
Is there a trick to using the MsiBreak environment variable to debug a CA?

### DISCUSSION THREAD

**Discussion by chrisbednarski** [wixtoolset/issues#7667] [Questions] [answered]
the `MsiBeak` environment variable can be used to debug CAs according to [here](https://learn.microsoft.com/en-us/windows/win32/msi/debugging-custom-actions)
I've also seen mentions of MMsiBreak for debugging managed CAs

I keep resolving to adding a message box and rebuilding my code to step into CAs

![image](https://github.com/wixtoolset/issues/assets/1804713/d11e55a0-f9ee-4622-b860-78894fdedc49)

I've tried setting `MsiBreak` as a system level environment variable and have not been able to make it work.

1. Has anyone managed to get this working?
2. What value should be set if I want to break on `Wix4AddMachineCertificate_X64` ?

**Comment by barnson** [comment, marked as answer]
Message boxes are state of the CA debugging art.

**Comment by robmen** [comment]
`AssertSz` exists in `dutil.h` for this very reason. The environment variables are finicky. I think it requires restarting the msiexec CA processes to get the environment variables to get picked up. Finicky is the best word I can use to describe it.
